test/integration/openapi2-test.js

```diff
@@ -0,0 +1,152 @@
+const { assert } = require('chai');
+
+const logger = require('../../lib/logger');
+const reporterOutputLogger = require('../../lib/reporters/reporterOutputLogger');
+const Dredd = require('../../lib/Dredd');
+
+const PORT = 9876;
+
+let output = '';
+
+function execCommand(options = {}, cb) {
+  output = '';
+  let finished = false;
+  if (!options.server) { options.server = `http://127.0.0.1:${PORT}`; }
+  if (!options.loglevel) { options.loglevel = 'warning'; }
+  new Dredd(options).run((error) => {
+    if (!finished) {
+      finished = true;
+      if (error ? error.message : undefined) {
+        output += error.message;
+      }
+      cb();
+    }
+  });
+}
+
+
+function record(transport, level, message) {
```

Comment:
Not critical, however, during the upgrade to `winston@3` I've noticed we declare this `record` function multiple times, doing the very same operations. I suggest to consider declaring just one `record` function (i.e. in some test utils) to decrease the refactoring surface of any future logging upgrades.

```js
// Convert to high-order function to preserve call signature of winston's callback.
function record(output) {
  return (transport, level, message) => {...}
}
```

Reply:
Agree 👍 I'd like to consolidate the ways logging gets inspected in tests together with further improving the logging infrastructure after it was split between logs and output.